internal/clients/interface.go

```diff
@@ -90,11 +90,14 @@ var GetServiceAzureClient func(ctx context.Context) (ServiceAzure, error)
 type Azure interface {
 	ClientStatuser
 
+	// EnsureResourceGroup makes sure that group with give name exists in a location
+	EnsureResourceGroup(ctx context.Context, name string, location string) (*string, error)
+
```

Comment:
Yeah bummer to do that for all steps, maybe you can figure out a solution (passing some interface in), but that really sounds clunky this is the cleanest approach I think.

Reply:
Yeah, I'll see how it goes with other steps, we might want to put some steps together, maybe "Prepare networking resources" etc :)